openpype/hosts/houdini/api/lib.py

```diff
@@ -293,46 +293,50 @@ def imprint(node, data):
     Args:
         node(hou.Node): node object from Houdini
         data(dict): collection of attributes and their value
+        update (bool, optional): flag if imprint should update
+            already existing data or leave them untouched and only
+            add new.
 
     Returns:
         None
 
     """
+    if not data:
+        return
 
-    parm_group = node.parmTemplateGroup()
+    current_parameters = node.spareParms()
+    current_keys = [p.name() for p in current_parameters]
+    update_keys = []
 
+    parm_group = node.parmTemplateGroup()
     parm_folder = hou.FolderParmTemplate("folder", "Extra")
+    templates = []
     for key, value in data.items():
         if value is None:
             continue
 
-        if isinstance(value, float):
-            parm = hou.FloatParmTemplate(name=key,
-                                         label=key,
-                                         num_components=1,
-                                         default_value=(value,))
-        elif isinstance(value, bool):
-            parm = hou.ToggleParmTemplate(name=key,
-                                          label=key,
-                                          default_value=value)
-        elif isinstance(value, int):
-            parm = hou.IntParmTemplate(name=key,
-                                       label=key,
-                                       num_components=1,
-                                       default_value=(value,))
-        elif isinstance(value, six.string_types):
-            parm = hou.StringParmTemplate(name=key,
-                                          label=key,
-                                          num_components=1,
-                                          default_value=(value,))
-        else:
-            raise TypeError("Unsupported type: %r" % type(value))
-
-        parm_folder.addParmTemplate(parm)
-
+        if key in current_keys:
+            if not update:
+                print(f"{key} already exists on {node}")
+            else:
+                print(f"replacing {key}")
+                update_keys.append((key, value))
+            continue
+        parm = parm_to_template(key, value)
+        # parm.hide(True)
+        templates.append(parm)
+    parm_folder.setParmTemplates(templates)
     parm_group.append(parm_folder)
     node.setParmTemplateGroup(parm_group)
 
+    if update_keys:
+        parms = node.parmTuplesInFolder(("Extra",))
+        for parm in parms:
+            for key, value in update_keys:
+                if parm.name() == key:
+                    node.replaceSpareParmTuple(
+                        parm.name(), parm_to_template(key, value))
+
```

Comment:
Why can't we just use the existing collected parms by name from above but instead need to look into `node.parmTuplesInFolder(("Extra",))` here?

Reply:
yeah, I've optimized all of it a little bit. Still wondering wheter update should be imprint or if it should be moved to separate function.